ovgs.proto

```diff
@@ -1,175 +1,241 @@
 // Copyright (c) 2023 Arista Networks, Inc.
```

Comment:
Thanks a lot. Looks nicer :)

There are a few things that aren't evident just by looking at the proto (they were mentioned in the interface doc that was shared). I'll list them down so we can document them here too, given they're part of the pubic interface and might be relevant when it comes to rolling an implementation of this (I've mentioned some of them through the other comments)- 

1.) viewUserRole/GetUserRole - A user can only view roles of another user in the groups that it has a role assigned to.
2.) GetOwnershipVoucher - Validates that the requested lifetime is not in the past (Do we need to return specific error code for this?)
3.) viewGroup/GetGroup - A root group with the name org_id is precreated and all serials purchased by that org are assigned to the root group.
4.) Clients must delete all subgroups/children before they can delete a group. Any pdc's and serials must be unassigned.

Reply:
feel free to add whatever text into this proto to help clarify these points 
you are welcome to commit to this pr

Reply:
I lack the permissions to commit to this -
harshitk@Harshits-MacBook-Pro ownership-voucher-grpc % git push mhines3 HEAD:main
remote: Permission to h-fam/ownership-voucher-grpc.git denied to harshitk-arista.
fatal: unable to access 'https://github.com/h-fam/ownership-voucher-grpc.git/': The requested URL returned error: 403

You can give me the permissions or if it's quicker, apply the patch yourself (both work) -
[0001-Add-more-comments-detailing-the-behavior-of-various-.patch](https://github.com/aristanetworks/ownership-voucher-grpc/files/10906654/0001-Add-more-comments-detailing-the-behavior-of-various-.patch)